hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -207,13 +219,21 @@ public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(name);
         out.writeInt(type.getId());
         writeNullableList(attributes, out);
+
+        if (type == IndexType.BITMAP) {
+            out.writeObject(bitmapIndexOptions);
+        }
     }
 
     @Override
     public void readData(ObjectDataInput in) throws IOException {
         name = in.readUTF();
         type = IndexType.getById(in.readInt());
         attributes = readNullableList(in);
+
+        if (type == IndexType.BITMAP) {
+            bitmapIndexOptions = in.readObject();
```

Comment:
If we do not write options unconditionally, we will not validate them on the server-side. Consider a user who submits `IndexConfig{type=HASH, bitmapOptions=<not null object>}`. This is an invalid config, but how do we know what was his real intention - to create a hash index or a bitmap index? 
I would rather read/write the object as is, and throw an exception on invalid config, rather than ignoring invalid state silently.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/16305/commits/123a22619c1c2ed7d26cabc286f88dbe801cff06